### ISSUE TITLE
braces rule adds newline before ?>

### ISSUE THREAD

**Issue by tillkruss** [FriendsOfPHP/PHP-CS-Fixer#1829]
The `braces` rule turns the following code:

```
<?php echo $foobar; ?>
```
Into:
```
<?php echo $foobar;
?>
```

Is that intended?

I'm running version: `1.11.2`

**Comment by tillkruss** [comment]
Upgrading from `1.10` to `1.11.2` fixed it.

**Comment by GrahamCampbell** [comment]
Are you sure the braces fixer is doing that?

**Comment by GrahamCampbell** [comment]
If it's not the braces fixer, please confirm that if you add multiple php open tags to the file, it doesn;t happen anymore.

**Comment by GrahamCampbell** [comment]
Say:

```php
<?php echo $foobar; ?>
<?php echo $foobar; ?>
```

**Comment by tillkruss** [comment]
Can you explain? I have loads of opening tags in the file. It's an old code base.

**Comment by tillkruss** [comment]
When I uncomment the `braces` fixer it doesn't happen.

When I enable the `braces` fixer I see that the following fixers have been executed:

```
unary_operators_spaces, logical_not_operators_with_successor_space, braces
```

**Comment by GrahamCampbell** [comment]
> Can you explain?

PHP fixer treats monolithic files differently. Feel free to flick through the code. It'll make more sense than me explaining it badly.

**Comment by SpacePossum** [comment]
I tried some stuff, like
```php
<?php echo 1; ?>
<?php echo 2; ?>
<?php echo 3; ?>
<?php echo 4; ?>
```

however I cannot reproduce the issue you report.
Could you maybe provide the following to help us track down the issue:
- minimal code sample that goes wrong
- the configuration you use

**Comment by tillkruss** [comment]
Sure, this is the smallest working example:

```
<?php

class Foo
{
    public function bar()
    {
        ?>

            <input type="checkbox" value="<?php echo $foobar; ?>"> Email

        <?php
    }
}

```

This is my configuration:

```
$fixers = [
    'blankline_after_open_tag',
    'braces',
    'concat_without_spaces',
    'double_arrow_multiline_whitespaces',
    'duplicate_semicolon',
    'elseif',
    'empty_return',
    'encoding',
    'eof_ending',
    'extra_empty_lines',
    'function_call_space',
    'function_declaration',
    'include',
    'indentation',
    'join_function',
    'line_after_namespace',
    'linefeed',
    'list_commas',
    'logical_not_operators_with_successor_space',
    'lowercase_constants',
    'lowercase_keywords',
    'method_argument_space',
    'multiline_array_trailing_comma',
    'multiline_spaces_before_semicolon',
    'multiple_use',
    'namespace_no_leading_whitespace',
    'no_blank_lines_after_class_opening',
    'no_empty_lines_after_phpdocs',
    'object_operator',
    'operators_spaces',
    'parenthesis',
    'phpdoc_indent',
    'phpdoc_inline_tag',
    'phpdoc_no_access',
    'phpdoc_no_package',
    'phpdoc_scalar',
    'phpdoc_short_description',
    'phpdoc_to_comment',
    'phpdoc_trim',
    'phpdoc_type_to_var',
    'phpdoc_var_without_name',
    'remove_leading_slash_use',
    'remove_lines_between_uses',
    'return',
    'self_accessor',
    'short_array_syntax',
    'short_echo_tag',
    'short_tag',
    'single_array_no_trailing_comma',
    'single_blank_line_before_namespace',
    'single_line_after_imports',
    'single_quote',
    'spaces_before_semicolon',
    'spaces_cast',
    'standardize_not_equal',
    'ternary_spaces',
    'trailing_spaces',
    'trim_array_spaces',
    'unalign_equals',
    'unary_operators_spaces',
    'unused_use',
    'visibility',
    'whitespacy_lines',
];

return Config::create()
    ->finder(DefaultFinder::create()->in(__DIR__))
    ->fixers($fixers)
    ->level(FixerInterface::NONE_LEVEL)
    ->setUsingCache(true);
```

**Comment by keradus** [comment]
fixed by #1840
